cloud: optimize TOC and add concept docs (#19717)
pingcap/docs: tidb-cloud/sql-concepts.md

- DDL (Data Definition Language): It is used to define database objects, including databases, tables, views, and indexes. For DDL statements in TiDB, see [Schema management / Data definition statements (DDL)](/sql-statements/sql-statement-overview.md#schema-management--data-definition-statements-ddl).

- DML (Data Manipulation Language): It is used to manipulate application-related records. For DML statements in TiDB, see [Data manipulation statements (DML)](/sql-statements/sql-statement-overview.md#data-manipulation-statements-dml).

- DQL (Data Query Language): It is used to query the records after conditional filtering.

- DCL (Data Control Language): It is used to define access privileges and security levels.

To get an overview of SQL statements in TiDB, see [SQL Statement Overview](/sql-statements/sql-statement-overview.md).

## SQL mode

TiDB servers operate in different SQL modes and apply these modes differently for different clients. SQL mode defines the SQL syntax that TiDB supports and the type of data validation check to perform.

For more information, see [SQL Mode](/sql-mode.md).

## Row ID generation attributes

> TiDB provides three SQL attributes to optimize row ID generation and data distribution to address performance and scalability challenges.

Author: ```suggestion
TiDB provides three SQL attributes to optimize row ID generation and data distribution.
```

For more information, see [AUTO_RANDOM](/auto-random.md).

### SHARD_ROW_ID_BITS

For the tables with a non-clustered primary key or no primary key, TiDB uses an implicit auto-increment row ID. When a large number of `INSERT` operations are performed, the data is written into a single Region, causing a write hot spot.

To mitigate the hot spot issue, you can configure [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md). The row IDs are scattered, and the data are written into multiple different Regions.

## Keywords

Keywords are words that have special meanings in SQL statements, such as `SELECT`, `UPDATE`, and `DELETE`.

- Some of them can be used as identifiers directly, which are called non-reserved keywords.

- Some of them require special treatment before being used as identifiers, which are called reserved keywords.

> However, there are special non-reserved keywords that might still require special treatment. It is recommended that you treat them as reserved keywords.

Reviewer: ```suggestion
However, some non-reserved keywords might still require special treatment. It is recommended that you treat them as reserved keywords.
```
